extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -1,55 +1,69 @@
 /**
  * External dependencies
  */
-import { get, startsWith } from 'lodash';
 import { __, sprintf } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 import { Button } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { get, startsWith } from 'lodash';
 import { withSelect } from '@wordpress/data';
 import { Warning } from '@wordpress/editor';
 import Gridicon from 'gridicons';
 
 /**
  * Internal dependencies
  */
+import analytics from 'lib/analytics';
 import getSiteFragment from '../get-site-fragment';
 import './store';
 
 import './style.scss';
 
-const UpgradeNudge = ( { planName, planPathSlug, postId, postType } ) => (
-	<Warning
-		actions={ [
-			<Button
-				href={ addQueryArgs(
-					`https://wordpress.com/checkout/${ getSiteFragment() }/${ planPathSlug }`,
-					{
-						redirect_to: `/${ postType }/${ getSiteFragment() }/${ postId }`,
-					}
-				) }
-				target="_top"
-				isDefault
+class UpgradeNudge extends Component {
+	trackUpgradeClick = () => {
```

Comment:
I changed this to be an arrow component without testing it. I don't believe it was correctly resolving `this.props` before, which likely led to `undefined` `planPathSlug` in the tracks event.